src/Components/Slider/Slider.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+let isFirstMount = true;
+let isTransitionEnd = true;
+
+const usePrevious = (value) => {
+  const lastDistance = useRef();
+  useEffect(() => {
+    lastDistance.current = value;
+  });
+  return lastDistance;
+};
+
+function Slider({ image }) {
+  const slideContainerRef = useRef();
+  const [distance, setDistance] = useState(-800);
+  const [images, setImages] = useState(image);
+  const [transitionTime, setTransitionTime] = useState('0.5s');
+  const lastDistance = usePrevious(distance).current;
```

Comment:
- usePrevious 훅에서 lastDistance.current 를 리턴하는 방향으로 작성해주시면, 더 통일성 있는 코드가 될 듯 합니다!
```suggestion
  const lastDistance = usePrevious(distance);
```

Reply:
네 알겠습니다!